src/DiscriSettingsWindow.cpp

```diff
@@ -0,0 +1,138 @@
+        discri_boxLayout->addWidget(label_width, vPos_settingLabels, hPos_width);
+
+        int hPos_include = 3;
+        QLabel *label_include = new QLabel("Include");
+        label_include->setAlignment(Qt::AlignCenter);
+        discri_boxLayout->addWidget(label_include, vPos_settingLabels, hPos_include);
+
+        // Display the channel settings themselves
+        int vPos_channel = 0;
+        for (int dc_id = 0; dc_id < m_interface.m_conditions->getNDiscriChannels(); dc_id++) {
+            DiscriChannel discriChannel;
+
+            vPos_channel = dc_id+1;
+
+            discriChannel.label = new QLabel("Discri channel "+QString::number(dc_id)+ " : ");
+            discri_boxLayout->addWidget(discriChannel.label, vPos_channel, 0);
+
+            discriChannel.included = new QCheckBox();
+            discriChannel.included->setStyleSheet("margin-left:50%; margin-right:50%;");
+            discriChannel.included->setChecked(m_interface.m_conditions->getDiscriChannelState(dc_id));
+            discri_boxLayout->addWidget(discriChannel.included, vPos_channel, hPos_include);
+
+            discriChannel.threshold = new QSpinBox();
+            discriChannel.threshold->setValue(m_interface.m_conditions->getDiscriChannelThreshold(dc_id));
+            discri_boxLayout->addWidget(discriChannel.threshold, vPos_channel, hPos_threshold);
+
+            discriChannel.width = new QSpinBox();
+            discriChannel.width->setMaximum(200);
+            discriChannel.width->setValue(m_interface.m_conditions->getDiscriChannelWidth(dc_id));
+            discri_boxLayout->addWidget(discriChannel.width, vPos_channel, hPos_width);
+            m_discriChannels.push_back(discriChannel);
+
+        }
+
+        // Display the number of channel required for a coincidence 
+
+        int majorityPosition = vPos_channel + 1;
+        QLabel *globalSettingsLabel = new QLabel("Global settings : ");
+        discri_boxLayout->addWidget(globalSettingsLabel, majorityPosition+1, 0);
+
+        QLabel *label_majority = new QLabel("Majority");
+        discri_boxLayout->addWidget(label_majority, majorityPosition, 1);
+        m_box_majority = new QSpinBox();
+        m_box_majority->setRange(0, m_interface.m_conditions->getNDiscriChannels());
+        m_box_majority->setWrapping(1);
+        m_box_majority->setValue(m_interface.m_conditions->getChannelsMajority());
+        m_box_majority->setAlignment(Qt::AlignCenter);
+        discri_boxLayout->addWidget(m_box_majority, majorityPosition+1, 1);
+
+        m_box->setLayout(discri_boxLayout);
+        discri_layout->addWidget(m_box);
+
+        // Button to propagate the settings
+        m_btn_propagate = new QPushButton("Propagate");
+        connect(m_btn_propagate, &QPushButton::clicked, this, &DiscriSettingsWindow::propagate);
+        // FIXME log discri settings into the condition log
+        connect(m_btn_propagate, &QPushButton::clicked, &m_interface, &Interface::updateConditionLog);
```

Comment:
Maybe we could add a connection so that the window gets closed when clicking "propagate"?